### PR TITLE
feat(Select): implement Select component

### PR DESCRIPTION
**Pull request by CernaianuMihai**
This PR is about implementing the Select component

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-237837

### CODE REVIEW

File: src/components/_components.scss [hunk 1]
```diff
@@ -10,3 +10,4 @@
 @import 'WebexMember/WebexMember';
 @import 'WebexMemberRoster/WebexMemberRoster';
 @import 'WebexRemoteMedia/WebexRemoteMedia';
+@import 'WebexSettings/WebexSettings';
```
Comment (cipak) [inline src/components/_components.scss:13]: we should not export the (internal) Select component

File: src/components/WebexSettings/Select.stories.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (cipak) [inline src/components/WebexSettings/Select.stories.js:1]: there is no need for a dedicated story for the Select component, we are not inviting developers to use it, it's just for internal use.

File: src/components/WebexSettings/Select.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-select {
+  select {
+    font-family: "CiscoSansTT Regular","Helvetica Neue",Helvetica,Arial,sans-serif;
```
Comment (cipak) [inline src/components/WebexSettings/Select.scss:3]: isn't the font family already set globally?

File: src/components/WebexSettings/Select.jsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+export default function Select({className, ...props}) {
+  const mainClasses = {
+    [`${WEBEX_COMPONENTS_CLASS_PREFIX}-select`]: true,
+    [className]: !!className,
+  };
+
+  return (
+    <div className={classNames(mainClasses)}>
+      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+      <select {...props} />
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:23]: maybe send the options as a prop as well?

File: src/components/WebexSettings/Select.stories.js (hunk 2 above)
Comment (CernaianuMihai) [reply src/components/WebexSettings/Select.stories.js:1]: Ok, I'll remove it.

File: src/components/WebexSettings/Select.scss (hunk 3 above)
Comment (CernaianuMihai) [reply src/components/WebexSettings/Select.scss:3]: Yes, it is. I'll remove this line.

File: src/components/_components.scss (hunk 1 above)
Comment (cipak) [reply src/components/_components.scss:13]: sorry, I read "export" instead of "import":)

File: src/components/WebexSettings/Select.jsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';
+
+/**
+ * Select component
+ *
+ * @param {object} props Data passed to the component
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:9]: use double space between arg name and description `props  Data passed...`

File: src/components/WebexSettings/Select.jsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ * @param {object[]} props.options Array of HMTL option elements
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:10]: they are not (yet) html option elements, they are just simple objects with 2 props: value and label.

I see that you only use value, but value usually is some id (eg: camera device id) so we need a label to display to the user.

Please document these 2 props in the jsdoc, eg:
```js
* @param {string} props.options[].value  The value of the option
```

File: src/components/WebexSettings/Select.jsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  return (
+    <div className={classNames(mainClasses)}>
+      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+      <select {...props}>
+        {
+          options?.map((option) => (
+            <option
+              value={option.value}
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:28]: you can put this on a single line

File: src/components/WebexSettings/Select.jsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+          ))
+        }
+      </select>
+    </div>
+  );
+}
+
+Select.propTypes = {
+  className: PropTypes.string,
+  options: PropTypes.arrayOf(PropTypes.shape({value: PropTypes.value})),
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:41]: nice. Please add the `label` as well.

File: src/components/WebexSettings/Select.jsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+ * @returns {object} JSX of the element
+ */
+export default function Select({className, options, ...props}) {
+  const mainClasses = {
+    [`${WEBEX_COMPONENTS_CLASS_PREFIX}-select`]: true,
+    [className]: !!className,
+  };
+
+  return (
+    <div className={classNames(mainClasses)}>
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:22]: if not 100% needed, please remove this `<div>` wrapper and put the class directly on the `<select>` element.

File: src/components/WebexSettings/Select.jsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';
+
+/**
+ * Select component
+ *
+ * @param {object} props  Data passed to the component
+ * @param {object[]} props.options  Array of option elements
+ * @param {string} props.options[].value  The value of the option
+ * @param {string} props.options[].label  The label of the option
+ * @param {string} props.className  Custom CSS class to apply
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:13]: put className before options, just like in the function's signature

File: src/components/WebexSettings/Select.jsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import {WEBEX_COMPONENTS_CLASS_PREFIX} from '../../constants';
+
+/**
+ * Select component
+ *
+ * @param {object} props  Data passed to the component
+ * @param {object[]} props.options  Array of option elements
```
Comment (cipak) [inline src/components/WebexSettings/Select.jsx:10]: Array of options, or List of options

**Review by cipak — APPROVED** [review]

**Pull request merged**